util/network-devp2p/src/ip_utils.rs

```diff
@@ -236,16 +236,15 @@ mod getinterfaces {
 				let sa: *const sockaddr_in6 = sa as *const sockaddr_in6;
 				let sa = & unsafe { *sa };
 				let (addr, port) = (sa.sin6_addr.s6_addr, sa.sin6_port);
-				let addr: [u16; 8] = unsafe { mem::transmute(addr) };
 				(IpAddr::V6(Ipv6Addr::new(
-					addr[0],
-					addr[1],
-					addr[2],
-					addr[3],
-					addr[4],
-					addr[5],
-					addr[6],
-					addr[7])),
+					((addr[1] as u16) << 8) | (addr[0] as u16),
```

Comment:
`addr` is actually a [u8; 16] so no need to transmute it but requires some shifting to merge two bytes into a word (two bytes)

Reply:
Test: https://play.rust-lang.org/?gist=b26a380a7d08aa4b781508fc6d9f895e&version=stable&mode=debug